lib/hanami/slice_registrar.rb

```diff
@@ -122,10 +112,27 @@ def load_slice(slice_name)
       register(slice_name, slice_class)
     end
 
+    # Finds the path to the slice's definition file, if it exists, in the following order:
+    #
+    # 1. `config/slices/[slice_name].rb`
+    # 2. `slices/[parent_slice_name]/config/[slice_name].rb` (unless parent is the app)
+    # 3. `slices/[slice_name]/config/slice.rb`
+    #
+    # If the slice is nested under another slice then it will look in the following order:
+    #
+    # 1. `config/slices/[parent_slice_name]/[slice_name].rb`
+    # 2. `slices/[parent_slice_name]/config/[slice_name].rb`
+    # 3. `slices/[parent_slice_name]/[slice_name]/config/slice.rb`
     def find_slice_require_path(slice_name)
+      app_slice_file_path = [slice_name]
+      app_slice_file_path.prepend(parent.slice_name) unless parent.eql?(parent.app)
```

Comment:
I think this won't work for deeply nested slices, right? So in the case of a slice structure like:

```
slices/top/slices/middle/slices/bottom/
```

Then this code would look for `config/slices/middle/bottom.rb`, when it should _actually_ be looking for `config/slices/top/middle/bottom.rb`, right?

Reply:
I was wrong. For a deeply nested slice (e.g. `Top::Middle::Bottom::Slice`), its parent will have a `slice_name.to_s` of `top/middle`, which means we look for the correct slice path inside `/config/slices/`. The extra specs I've added in dc71e89 demonstrate this.